src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheck.java

```diff
@@ -37,8 +37,12 @@
     extends AbstractViolationReporter
     implements FileSetCheck {
 
-    /** Collects the error messages. */
-    private final SortedSet<LocalizedMessage> messageCollector = new TreeSet<>();
+    /**
+     * Collects the error messages.
+     * @noinspection ThreadLocalNotStaticFinal
+     */
+    private final ThreadLocal<SortedSet<LocalizedMessage>> messageCollector =
```

Comment:
@romani , please, take a look at my explanations to @rnveach 

> Suppose we have two instances (named `foo` and `bar`) of some `FileSet` class. We call `foo.log(...)` and then `foo::messageCollector` contains one message. However, after calling `bar.log(...)` the `foo::messageCollector` will contain two messages, because the field is static and we're inside the same thread. Not sure if this is the expected behavior.

If we have a static field, then all checks in the same thread will have the same list of messages.